packages/server/modules/serverinvites/services/processing.ts

```diff
@@ -210,7 +293,7 @@ export const finalizeResourceInviteFactory =
 
     try {
       // Add email
-      if (isNewEmailTarget) {
+      if (isNewEmailTarget && action === InviteFinalizationAction.ACCEPT) {
```

Comment:
There was a bug here where new email was added even if action was DECLINE (new test case added for this too)